### DISCUSSION TITLE
issue with tweaks when placed under a category

### DISCUSSION THREAD

**Discussion by iamLazyCode** [moudey/Shell#212] [General]
hi @moudey i have made atweak to restart explorer and placed under context menu

when its not placed under any category it works fine like this
<img width="426" alt="Screenshot_10" src="https://user-images.githubusercontent.com/13099434/220441833-8f6f62fe-6b8a-4d47-9fd5-406c912ad384.png">

but when i place it under a category with 
item(find='restart exlorer ' pos=0 sep='bottom' menu="Misc Tools")

the further menu doesnt appear

<img width="502" alt="Screenshot_9" src="https://user-images.githubusercontent.com/13099434/220442104-f4de267d-f8e3-44d7-828f-e2e2a40d369d.png">

---------------------------------------------------------------------------------

Also I am trying to remove vlc 
![image](https://user-images.githubusercontent.com/13099434/220444791-9be6c37c-457b-41bc-b8bd-152711d93f49.png)

commands i used 
```
item(find='add to vlc media player's playlist' find='add to vlc media player's playlist' vis=vis.remove)
item(find='play with vlc media player' find='play with vlc media player' vis=vis.remove)
```

but still they are intact...thankyou very much

**Comment by moudey** [comment]
Is this "restart explorer" item static or dynamic?

---
>Also I am trying to remove vlc

Use double quotes if the text contains single quotes or vice versa

```
item(find="add to vlc media player's playlist" vis=vis.remove)
item(find="play with vlc media player" vis=vis.remove)
```

Or use `|` sign. To search for more than one phrase
```
item(find="add to vlc media player's playlist | play with vlc media player" vis=vis.remove)
```
Or through this command it removes all items that contain "vlc media player".
```
item(find='vlc media player' vis=vis.remove)
```

**Comment by iamLazyCode** [comment]
> Is this "restart explorer" item static or dynamic?
> 
> > Also I am trying to remove vlc
> 
> Use double quotes if the text contains single quotes or vice versa
> 
> ```
> item(find="add to vlc media player's playlist" vis=vis.remove)
> item(find="play with vlc media player" vis=vis.remove)
> ```
> 
> Or use `|` sign. To search for more than one phrase
> 
> ```
> item(find="add to vlc media player's playlist | play with vlc media player" vis=vis.remove)
> ```
> 
> Or through this command it removes all items that contain "vlc media player".
> 
> ```
> item(find='vlc media player' vis=vis.remove)
> ```

i am not sure if its dynamic or static ...how can i check this sir ?

also when i right click on a folder there are shortcuts for even apps that i uninstalled...i think thats creating issue with removinf the vlc
anyhow i can reset the context menu sir ?

**Comment by moudey** [comment]
This item appears on the desktop only, so we define the type and search by matching the name of the item with double quotes inside `find` property
```
item(type='desktop' find='"restart explorer"' pos=0 sep='bottom' menu="Misc Tools")
```

**Comment by iamLazyCode** [comment]
> 

you are simply the best sir , amazingggg :)


aah same another one is

item(type='desktop' find='"system file checker"' pos=0 sep='bottom' menu="Misc Tools")

the further menu doesnt opens sir

**Comment by moudey** [comment]
This command is correct and should be work. Restart the explorer to load the settings changes

**Comment by iamLazyCode** [comment]
> This command is correct and should be work. Restart the explorer to load the settings changes

it surely worked for restart explorer but not working for the system file checker


edit : 
i guess its the issue with my windows...il reinstall and update u sir

**Comment by moudey** [comment]
You can add these commands directly to **Shell** instead of adding them to the registry
```
menu(title='Restart Explorer' image='explorer.exe')
{
    item(title='Restart Explorer Now' cmd args='/c taskkill /f /im explorer.exe & start explorer.exe')
    item(title='Restart Explorer with Pause' cmd args='/c @@echo off & echo. & echo Stopping explorer.exe process . . . & echo. & taskkill /f /im explorer.exe & echo. & echo. & echo Waiting to start explorer.exe process when you are ready . . . & pause && start explorer.exe && exit')
}

menu(title='System File Checker' image='WmiPrvSE.exe')
{
    tem(title='Run System File Checker' admin=true cmd='PowerShell' args="-windowstyle hidden -command \"Start-Process cmd -ArgumentList '/s,/k, sfc /scannow' -Verb runAs\"")
    item(title='System File Checker log' cmd='PowerShell' args='(sls [SR] $env:windir\Logs\CBS\CBS.log -s).Line >"$env:userprofile\Desktop\sfcdetails.txt\"')
}
```

**Comment by iamLazyCode** [comment]
> You can add these commands directly to **Shell** instead of adding them to the registry
> 
> ```
> menu(title='Restart Explorer' image='explorer.exe')
> {
>     item(title='Restart Explorer Now' cmd args='/c taskkill /f /im explorer.exe & start explorer.exe')
>     item(title='Restart Explorer with Pause' cmd args='/c @@echo off & echo. & echo Stopping explorer.exe process . . . & echo. & taskkill /f /im explorer.exe & echo. & echo. & echo Waiting to start explorer.exe process when you are ready . . . & pause && start explorer.exe && exit')
> }
> 
> menu(title='System File Checker' image='WmiPrvSE.exe')
> {
>     tem(title='Run System File Checker' admin=true cmd='PowerShell' args="-windowstyle hidden -command \"Start-Process cmd -ArgumentList '/s,/k, sfc /scannow' -Verb runAs\"")
>     item(title='System File Checker log' cmd='PowerShell' args='(sls [SR] $env:windir\Logs\CBS\CBS.log -s).Line >"$env:userprofile\Desktop\sfcdetails.txt\"')
> }
> ```

sir restart explorer works great but there is something wrong in system file checker it sets the conext menu back to normal

**Comment by moudey** [comment]
>sir restart explorer works great but there is something wrong in system file checker it sets the conext menu back to normal

Sorry 
```
menu(title='System File Checker' image='WmiPrvSE.exe')
{
    item(title='Run System File Checker' admin=true cmd='PowerShell' args="-windowstyle hidden -command \"Start-Process cmd -ArgumentList '/s,/k, sfc /scannow' -Verb runAs\"")
    item(title='System File Checker log' cmd='PowerShell' args='(sls [SR] $env:windir\Logs\CBS\CBS.log -s).Line >"$env:userprofile\Desktop\sfcdetails.txt\"')
}
```

**Comment by iamLazyCode** [comment]
> > sir restart explorer works great but there is something wrong in system file checker it sets the conext menu back to normal
> 
> Sorry
> 
> ```
> menu(title='System File Checker' image='WmiPrvSE.exe')
> {
>     item(title='Run System File Checker' admin=true cmd='PowerShell' args="-windowstyle hidden -command \"Start-Process cmd -ArgumentList '/s,/k, sfc /scannow' -Verb runAs\"")
>     item(title='System File Checker log' cmd='PowerShell' args='(sls [SR] $env:windir\Logs\CBS\CBS.log -s).Line >"$env:userprofile\Desktop\sfcdetails.txt\"')
> }
> ```

ur the best sir works flawlessly thankyou so much sir ... my pay day is near for some coffees <3

**Comment by iamLazyCode** [comment]
> > sir restart explorer works great but there is something wrong in system file checker it sets the conext menu back to normal
> 
> Sorry
> 
> ```
> menu(title='System File Checker' image='WmiPrvSE.exe')
> {
>     item(title='Run System File Checker' admin=true cmd='PowerShell' args="-windowstyle hidden -command \"Start-Process cmd -ArgumentList '/s,/k, sfc /scannow' -Verb runAs\"")
>     item(title='System File Checker log' cmd='PowerShell' args='(sls [SR] $env:windir\Logs\CBS\CBS.log -s).Line >"$env:userprofile\Desktop\sfcdetails.txt\"')
> }
> ```

one last thing sir

is it possible to move these under like this
![image](https://user-images.githubusercontent.com/13099434/220479584-3583388f-d148-42f0-b1d4-c9cdabe9bed6.png)

**Comment by moudey** [comment]
Paste these items inside the "Misc Tools" items as in the following example:
```
dynamic
{
    ...
    menu(title="Misc Tools")
    {
        menu(title='Restart Explorer' image='explorer.exe')
        {
            item(title='Restart Explorer Now' cmd args='/c taskkill /f /im explorer.exe & start explorer.exe')
            item(title='Restart Explorer with Pause' cmd args='/c @@echo off & echo. & echo Stopping explorer.exe process . . . & echo. & taskkill /f /im explorer.exe & echo. & echo. & echo Waiting to start explorer.exe process when you are ready . . . & pause && start explorer.exe && exit')
        }

        menu(title='System File Checker' image='WmiPrvSE.exe')
        {
            item(title='Run System File Checker' admin=true cmd='PowerShell' args="-windowstyle hidden -command \"Start-Process cmd -ArgumentList '/s,/k, sfc /scannow' -Verb runAs\"")
            item(title='System File Checker log' cmd='PowerShell' args='(sls [SR] $env:windir\Logs\CBS\CBS.log -s).Line >"$env:userprofile\Desktop\sfcdetails.txt\"')
        }
    }
}
```

Don't forget to delete these items from the registry to avoid duplicating them
```
reg delete "HKEY_CLASSES_ROOT\DesktopBackground\Shell\Restart Explorer" /f
reg delete "HKEY_CLASSES_ROOT\DesktopBackground\Shell\SFC" /f
```

**Comment by iamLazyCode** [comment]
> Paste these items inside the "Misc Tools" items as in the following example:
> ```
> dynamic
> {
>     ...
>     menu(title="Misc Tools")
>     {
>         menu(title='Restart Explorer' image='explorer.exe')
>         {
>             item(title='Restart Explorer Now' cmd args='/c taskkill /f /im explorer.exe & start explorer.exe')
>             item(title='Restart Explorer with Pause' cmd args='/c @@echo off & echo. & echo Stopping explorer.exe process . . . & echo. & taskkill /f /im explorer.exe & echo. & echo. & echo Waiting to start explorer.exe process when you are ready . . . & pause && start explorer.exe && exit')
>         }
> 
>         menu(title='System File Checker' image='WmiPrvSE.exe')
>         {
>             item(title='Run System File Checker' admin=true cmd='PowerShell' args="-windowstyle hidden -command \"Start-Process cmd -ArgumentList '/s,/k, sfc /scannow' -Verb runAs\"")
>             item(title='System File Checker log' cmd='PowerShell' args='(sls [SR] $env:windir\Logs\CBS\CBS.log -s).Line >"$env:userprofile\Desktop\sfcdetails.txt\"')
>         }
>     }
> }
> ```
> 
> Don't forget to delete these items from the registry to avoid duplicating them
> ```
> reg delete "HKEY_CLASSES_ROOT\DesktopBackground\Shell\Restart Explorer" /f
> reg delete "HKEY_CLASSES_ROOT\DesktopBackground\Shell\SFC" /f
> ```

Will try and update u sir thankyou so much again

**Comment by iamLazyCode** [comment]
@moudey sir i am trying to add a exe file in shell

`item(title='app'  "Z:/Pc Stuff/Install/More Apps/Must Have Free Apps and Alternatives/Keyboard.exe"')`

what argument i need to pass to make it work ?

**Comment by moudey** [comment]
You forgot to add the command property
```
item(title='app' cmd='Z:/Pc Stuff/Install/More Apps/Must Have Free Apps and Alternatives/Keyboard.exe')
```

**Comment by iamLazyCode** [comment]
> cmd=

thankyou sir , i am assigning icons but somehow they are acting strange

this happens only when i use like this and assign a color to icon
`image=[\uE22B, #f00]`

<img width="262" alt="Screenshot_1" src="https://user-images.githubusercontent.com/13099434/221394082-ac17a770-b5ca-4c9d-aaa2-cbff0e9fdff8.png">


<img width="334" alt="Screenshot_2" src="https://user-images.githubusercontent.com/13099434/221394087-e0a08425-2212-416e-be3f-d7160a1fd8eb.png">


<img width="274" alt="Screenshot_4" src="https://user-images.githubusercontent.com/13099434/221394091-1c0b6657-67cd-437d-8fe7-8f57ec7eea64.png">


<img width="338" alt="Screenshot_3" src="https://user-images.githubusercontent.com/13099434/221394097-b5a4e007-ec9a-486c-907e-39a21f585572.png">

**Reply by moudey** [reply]
Hi @ifelixculpa  
This issue has been fixed. You can test it with this [build](https://nilesoft.org/download/shell/debug.zip)

**Reply by iamLazyCode** [reply]
> Hi @ifelixculpa  
> This issue has been fixed. You can test it with this [build](https://nilesoft.org/download/shell/debug.zip)

Awesome will test after i reach home sir... and i think still no update on chocolatey ? These dudes i think have trillions of pending stuff

**Reply by iamLazyCode** [reply]
> Hi @ifelixculpa This issue has been fixed. You can test it with this [build](https://nilesoft.org/download/shell/debug.zip)

thanks alot sir it works but i dont know why whole panel turned black

![image](https://user-images.githubusercontent.com/13099434/230303816-e3c28d25-ff22-4b36-854f-bb339a601a8a.png)

**Reply by moudey** [reply]
This is a test version. You can modify the theme settings or remove them from the configuration file to apply the default sttings.

**Reply by iamLazyCode** [reply]
> This is a test version. You can modify the theme settings or remove them from the configuration file to apply the default sttings.

Its ok sir i have to reinstall windows none the less will do it when this version is finLly released

**Comment by moudey** [comment]
What type of image (png, ico, svg)?
Did you change the `font` or `image` size in `shell.nss`?
Did you change the display scale?

**Comment by iamLazyCode** [comment]
> What type of image (png, ico, svg)? Did you change the `font` or `image` size in `shell.nss`? Did you change the display scale?

icon is basically from the repo u have posted sir ...
this is the shell script haven't touched many things as windows is fresh... also i just noticed another thing

see icon on desktop

<img width="331" alt="Screenshot_3" src="https://user-images.githubusercontent.com/13099434/221402724-79847356-d2b1-4285-a494-9fddee982314.png">


see icon on explorer right click
<img width="836" alt="Screenshot_2" src="https://user-images.githubusercontent.com/13099434/221402745-a9b335cd-04f9-48ac-8b0c-9cea2197e461.png">


```
shell
{
	var
	{
		isw11=sys.is11
	}
	
	set
	{
		theme
		{
			name="modern"
			background
			{
				//opacity=85
				//effect=1
			}
			image.align = 2// 0 = checked, 1 = image, 2 = both
		}

		tip
		{
			enabled=1
			opacity=100
			width=400
			radius=1
			time=1.25
			padding=[10,10]
		}

		exclude.where = !process.is_explorer
		showdelay=200
	}

	images import 'imports/images.nss'

	static
	{	
		import 'imports/static.nss'
		item(where=this.title.length > 25 menu=title.more_options)
	}

	dynamic
	{
		menu(mode="multiple" title="Pin/Unpin" image=icon.pin) { }
		menu(mode="multiple" title=title.more_options image=icon.more_options) { }
		item(title='app'  image=[\uE22B, #f00] cmd='Z:\Pc_blocker.exe')
		
		import 'imports/terminal.nss'
		import 'imports/file-manage.nss'
		import 'imports/develop.nss'
		import 'imports/goto.nss'
		import 'imports/taskbar.nss'
	}
}
```

**Comment by iamLazyCode** [comment]
@moudey its still shows over sized icons sir

**Comment by moudey** [comment]
Did you change the display scale?

**Comment by iamLazyCode** [comment]
> Did you change the display scale?



> Did you change the display scale?

i just saw the display drivers set scale to 150% as recommeded for my display as on 100% icons are too small and hard to see

![image](https://user-images.githubusercontent.com/13099434/222137715-d94329a5-7aaa-4562-bc9a-266c7eae4f64.png)

**Comment by moudey** [comment]
It may work fine after pressing `CTRL` + `RIGHT-CLICK`

**Comment by iamLazyCode** [comment]
> It may work fine after pressing `CTRL` + `RIGHT-CLICK`

yes works this way but isnt there any fix to this sir ?

**Comment by moudey** [comment]
I will fix this issue soon

**Comment by iamLazyCode** [comment]
> I will fix this issue soon

thankyou so much sir , have a greta day

**Comment by iamLazyCode** [comment]
> I will fix this issue soon

@moudey sir one quick question

i need to always run a script with admin privilidges...is there a way to do ? either in script or always making the cmd run as admin when i run this script

**Comment by moudey** [comment]
Set the `admin` property to 1 to always execute the command as Administrator.
```
admin=1
```

**Comment by iamLazyCode** [comment]
> Set the `admin` property to 1 to always execute the command as Administrator.
> ```
> admin=1
> ```

Where do I add this admin = 1


item(title='cache clean' image=\uE1A8 pos=3 sep='bottom' menu="My Scripts" cmd args='/k "D:\cache.bat"')

**Comment by moudey** [comment]
```
item(title='cache clean' image=\uE1A8 pos=3 sep='bottom' menu="My Scripts" admin=1 cmd args='/k "D:\cache.bat"')
```

**Comment by iamLazyCode** [comment]
@moudey hi sir i ran into another confusion i hope u can help....when i select single file the options are all available but when i select multiple files all options are gone

![Screenshot_4](https://github.com/moudey/Shell/assets/13099434/7d423677-c536-465c-9798-2b1119f00acb)
![Screenshot_3](https://github.com/moudey/Shell/assets/13099434/995eac83-4f41-42ae-99ee-2a9b599beaf7)

**Comment by moudey** [comment]
Please attach the configuration file

**Reply by iamLazyCode** [reply]
hi thanks sir here it is

[shell.zip](https://github.com/moudey/Shell/files/11750910/shell.zip)

**Comment by moudey** [comment]
Add a `mode` property to the item and assign it a `multiple` value
```
menu(mode="multiple" title="Misc Tools" image=\uE0F6)
{
}
```

**Reply by iamLazyCode** [reply]
hi sir sorry i am confused could u edit the above file for "bulk rename here" please

**Comment by moudey** [comment]
[Modified configuration file](https://github.com/moudey/Shell/files/11761189/shell.zip). Improved removal of items, making them in one line, and moving them to the front of the static items section.

**Reply by iamLazyCode** [reply]
thankyou sooo much sir works flawless... will be getting a coffee ur way asap :)

**Comment by iamLazyCode** [comment]
@moudey a coffe your way sir :)

il keep them coming 🗡️ 

sir one question

To get all these options under one drive 


![Screenshot_1](https://github.com/moudey/Shell/assets/13099434/64e4e022-8326-4880-8f9c-fa61e4011e55)


is this the correct way ? and if so what shall i search and place there


![Screenshot_2](https://github.com/moudey/Shell/assets/13099434/9350feb8-1c5c-4df0-bc5e-c98003ecab21)

**Reply by moudey** [reply]
Thanks for your coffee.

```
modify(find='always keep on this device|free up space|copy link|manage access|view online' menu="/One drive")
```

**Reply by iamLazyCode** [reply]
thanks for the command but i think there is some error shell no longer loads after this

![Screenshot_3](https://github.com/moudey/Shell/assets/13099434/217c074f-4b81-45f0-9b41-704a321ede8a)

**Reply by moudey** [reply]
debug version

```
modify(find='always keep on this device|free up space|copy link|manage access|view online' menu="/One drive")
```

version 1.8.1
```
item(find='always keep on this device|free up space|copy link|manage access|view online' menu="/One drive")
```

**Reply by iamLazyCode** [reply]
sir though those options dont appear but there's no one drive option either

![image](https://github.com/moudey/Shell/assets/13099434/c56b6063-565b-4b35-bed6-2a7004f8d17e)

**Reply by moudey** [reply]
try this
```
item(find='always keep on this device|free up space|copy link|manage access|view online' parent="One drive")
```

**Reply by iamLazyCode** [reply]
still same sir

**Comment by iamLazyCode** [comment]
@moudey any help sir ?

**Comment by moudey** [comment]
> sir though those options dont appear but there's no one drive option either

If there are no items in the menu, they are removed by default

**Reply by iamLazyCode** [reply]
they got removed after this

`item(find='always keep on this device|free up space|copy link|manage access|view online' parent="One drive")`

**Reply by moudey** [reply]
Put it in the static section

**Reply by iamLazyCode** [reply]
> Put it in the static section

It was placed under static  sir

**Reply by moudey** [reply]
Which version are you using?

**Reply by iamLazyCode** [reply]
> Which version are you using?

Official one from chocolatey sir
